core/src/main/scala/za/co/absa/spline/harvester/dispatcher/KafkaLineageDispatcher.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.dispatcher
+
+import org.apache.commons.configuration.Configuration
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+import org.apache.kafka.common.header.Header
+import org.apache.spark.internal.Logging
+import za.co.absa.spline.harvester.dispatcher.httpdispatcher.ProducerApiVersion
+import za.co.absa.spline.harvester.dispatcher.kafkadispatcher.{KafkaLineageDispatcherConfig, _}
+import za.co.absa.spline.harvester.json.HarvesterJsonSerDe.impl._
+import za.co.absa.spline.producer.model.v1_1.{ExecutionEvent, ExecutionPlan}
+
+import scala.collection.JavaConverters.asJavaIterableConverter
+import scala.util.control.NonFatal
+
+/**
+ * KafkaLineageDispatcher is responsible for sending the lineage data to spline gateway through kafka
+ */
+class KafkaLineageDispatcher(config: KafkaLineageDispatcherConfig)
```

Comment:
It's Ok, but just wondering why have you introduced a _parameter object_ here? There are just three independent arguments. The other dispatchers define theirs params directly in the constructor and it looks sufficient and straightforward to me. Do you have another opinion or you plan adding more parameters to the class?

Reply:
I thought there will be more properties, but since producer have its own subconfig it ended up being only two for now.

I will remove it.